src/db.c

```diff
@@ -304,6 +304,7 @@ int getKeySlot(sds key) {
      * the key slot would fallback to calculateKeySlot.
      */
     if (server.current_client && server.current_client->slot >= 0 && server.current_client->flags & CLIENT_EXECUTING_COMMAND) {
+        serverAssertWithInfoDebug(server.current_client, NULL, calculateKeySlot(key)==server.current_client->slot);
```

Comment:
This helps bring in additional layer of validation for the correct slot being cached or not. This would be an expensive operation during production run(s) so we want only to be enabled during debugging/testing.